shotover-proxy/src/tcp.rs

```diff
@@ -0,0 +1,19 @@
+use anyhow::{anyhow, Result};
+use std::time::Duration;
+use tokio::{
```

Comment:
Could we rename `tls.rs` to `transport.rs` and put all the TCP/TLS stuff in there?

Reply:
IMO tls has enough going on to be its own module.
Would you want to do transport/tls and trasnsport/tcp ?

Reply:
No not that fussed. Just a thought. Can merge as is.